### DISCUSSION TITLE
Text editor startup time comparison

### DISCUSSION THREAD

**Discussion by gapan** [unixwork/xnedit#144] [General]
Hi, I thought you might wanted to see this. It compares startup times of several text editors, including xnedit.

https://blog.salixos.org/post/choosing-the-default-text-editor/

Relevant quotes:

> [Starting with a blank document] It’s amazing how much faster xnedit is, compared to everything else! It blows everything else away and the difference is definitely perceptible. Launching xnedit feels (and is) almost instant. Sure, Motif is an older toolkit and you may have expected it to be somewhat faster, but xnedit is loaded with features. I wasn’t expecting such a huge difference. If you don’t mind the old-school looks, this is the editor to use if you care about speed. It would certainly be ideal for older systems, but again, you can feel the difference in speed with newer systems too.

> [Opening a small file] Xnedit is still very fast, the fastest one after Leafpad and I’m still very impressed! It is also a bit faster in loading an existing file, instead of starting a new one, but the difference is not as pronounced as with the L*afpads.

> [Opening a big file with some issues] The L*afpads are the fastest! All editors are slower in loading the big file than the smaller one. Xnedit especially takes a toll here. I’m thinking that this may be due to the encoding errors that the big file has. Featherpad is also a lot slower than it was, it is now the 3rd slowest, after Pluma and Gedit.

**Comment by unixwork** [comment]
Hi, thanks for the link, this is really interesting.

A fast startup time is definitely important for me, so maintaining the speed and not destroying it with stupid features is a top priority.

Support for larger files is still not great in xnedit, however recently I improved the loading speed significantly. The problem is actually line wrapping, which is implemented really inefficiently.
